flytekit/remote/remote.py

```diff
@@ -147,14 +148,25 @@ def _get_git_repo_url(source_path):
     Get git repo URL from remote.origin.url
     """
     try:
-        from git import Repo
+        git_config = source_path / ".git" / "config"
+        if not git_config.exists():
+            raise ValueError(f"{source_path} is not a git repo")
```

Comment:
I added tests to check these code paths. It's strange that codecov is not picking it up.